test/core/security/credentials_test.cc

```diff
@@ -449,126 +449,141 @@ static void test_oauth2_token_fetcher_creds_parsing_missing_token_lifetime(
   grpc_http_response_destroy(&response);
 }
 
-typedef struct {
-  const char* key;
-  const char* value;
-} expected_md;
+namespace {
 
-typedef struct {
-  grpc_error_handle expected_error;
-  const expected_md* expected;
-  size_t expected_size;
-  grpc_credentials_mdelem_array md_array;
-  grpc_closure on_request_metadata;
-  grpc_call_credentials* creds;
-  grpc_polling_entity pollent;
-} request_metadata_state;
-
-static void check_metadata(const expected_md* expected,
-                           grpc_credentials_mdelem_array* md_array) {
-  for (size_t i = 0; i < md_array->size; ++i) {
-    size_t j;
-    for (j = 0; j < md_array->size; ++j) {
-      if (0 ==
-          grpc_slice_str_cmp(GRPC_MDKEY(md_array->md[j]), expected[i].key)) {
-        GPR_ASSERT(grpc_slice_str_cmp(GRPC_MDVALUE(md_array->md[j]),
-                                      expected[i].value) == 0);
-        break;
-      }
-    }
-    if (j == md_array->size) {
-      gpr_log(GPR_ERROR, "key %s not found", expected[i].key);
-      GPR_ASSERT(0);
+class RequestMetadataState {
+ public:
+  static RequestMetadataState* NewInstance(
```

Comment:
Actually, I just noticed that this factory function isn't actually necessary; instead, you can just initialize the `grpc_polling_entity` right in the ctor.